static/src/javascripts/projects/common/modules/crosswords/hidden-input.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import { createClass, createElement, findDOMNode } from 'react/addons';
+import bonzo from 'bonzo';
+import fastdom from 'fastdom';
+import { scrollTo } from 'lib/scroller';
+import { isBreakpoint } from 'lib/detect';
+
+const HiddenInput = createClass({
+    getInitialState() {
+        return {
+            value: this.props.value,
+        };
+    },
+
+    componentDidUpdate() {
+        if (
+            isBreakpoint({
+                max: 'mobile',
+            })
+        ) {
+            fastdom.read(() => {
+                const offsets = bonzo(findDOMNode(this.refs.input)).offset();
```

Comment:
Is there a reason to use bonzo over .offsetX

Reply:
[`offset()`](https://github.com/ded/bonzo/blob/fdfe965179754a9121c0ce3b0defec64e96934ae/src/bonzo.js#L785-L792) gets the `top` and `left` position of the element relative to the document, using a combination of `getClientBoundingRect`, `pageXOffset ` and `pageYOffset ` and `clientTop` and `clientLeft`. The `width` and height` values come from `offsetWidth` and `offsetHeight`.

I'd advise against refactoring for now while there are no tests in place, although it would be good to get rid of `bonzo` eventually.